operation/fhir-operation-validate/src/main/java/org/linuxforhealth/fhir/operation/validate/ValidateOperation.java

```diff
@@ -131,20 +120,66 @@ private OperationOutcome buildResourceValidOperationOutcome(List<Issue> issues)
     
     
     /**
-     * Check if a resource validation mode code is valid.
+     * This method does the following validations.
+     * 1. Validate if a resource validation mode code is valid.
+     * 2. Validate an interaction for a specified resource type.
+     * 3. Validate if persistence layer implementation supports update/create mode if mode = update/create.
+     * 4. Validate if modes update and delete are only be used when the operation is invoked at the resource instance level
      *
      * @param modeParameter resource validation mode code to be validated
+     * @param resourceHelper Resource operation provider for loading related Library resources
+     * @param resourceType a valid resource type string
+     * @param operationContext the FHIROperationContext associated with the request
      * @throws FHIROperationException when the resource validation mode code is invalid
      */
-    private void validateModeParameter(Parameter modeParameter) throws FHIROperationException {
+    private void validateModeParameter(ModeType modetype, FHIRResourceHelpers resourceHelper, String resourceType, FHIROperationContext operationContext) throws FHIROperationException {
+        if (modetype != null) {
+            // Validate an interaction for a specified resource type.
+            if ((modetype == ModeType.CREATE  
+                    || modetype == ModeType.UPDATE 
+                    || modetype == ModeType.DELETE)
+                    && resourceType != null) {
+                FHIRResourceHelpers.Interaction interaction = FHIRResourceHelpers.Interaction.from(modetype.value());
+                resourceHelper.validateInteraction(interaction, resourceType);
+            }
+            // Validate if persistence layer implementation supports update/create mode if mode = update/create.
```

Comment:
its a little confusing, but I think we should only perform this check (`isUpdateCreateEnabled`) if:
A. mode=create; or
B. mode=update but the resource doesn't exist yet

if the resource already exists, and mode=update, and the update interaction is allowed for this resource type, then the request is valid when the resource that is passed is valid (even if "isUpdateCreateEnabled" is false)

Reply:
if B is too complicated (because it requires reading of the resource), we could consider checking just A here

Reply:
This has been fixed as suggested.
we should only perform this check (isUpdateCreateEnabled) if
A. mode=create; or
B. mode=update but the resource doesn't exist yet